core/src/services/webhdfs/backend.rs

```diff
@@ -177,6 +180,22 @@ pub struct WebhdfsBackend {
 }
 
 impl WebhdfsBackend {
+    #[allow(dead_code)]
+    pub(crate) fn new(root: &str, endpoint: &str, auth: Option<String>) -> Result<Self> {
```

Comment:
We can remove this API if we don't use it.

Reply:
I think we might still need it since the [`services/webhdfs/message.rs`](https://github.com/morristai/incubator-opendal/blob/3f044d6d568eb0761c5cb56fd9393bc7317ef48f/core/src/services/webhdfs/message.rs#L186) uses it in the test. If we don't add `allow(dead_code)`, the compiler will show a warning.


![Screenshot 2023-06-20 161524](https://github.com/apache/incubator-opendal/assets/10728152/907dcad8-a568-44a7-a745-cf471d719879)

Reply:
It's wrong that trying to build backend in unit test. In `test_list_status` alike, we should check the value of `FileStatusWrapper` directly.

Reply:
Fixed. Sorry, it was my oversight.

Reply:
Let's remove `Backend::new`

Reply:
I want to point out why I'm doing this in the beginning. In the original `test_list_status` case, it uses a pager to test the `next` behaviour. My initial thought was to keep it intact, but since I added a backend to the `pager` struct, I have to add a new backend to the existing `pager` to test the `next` method. I guess I should ask you whenever I feel unsure about changing the origin code.